### ISSUE TITLE
[Subtask] support UpdateColumnPosition in Altertable for spark connector

### ISSUE THREAD

**Issue by caican00** [datastrato/gravitino#2494]
### Describe the subtask

please add related code to com.datastrato.gravitino.spark.catalogGravitinoCatalog.java, transfrom Spark TableChange to Gravitino TableChange, you could refer to [#2458](https://github.com/datastrato/gravitino/pull/2458)

```
  public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchTableException {
    throw new NotSupportedException("Doesn't support altering table for now");
  }
```

Related Spark SQL should add to SparkIT.java, please refer https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-alter-table.html

### Parent issue

https://github.com/datastrato/gravitino/issues/1227

**Comment by caican00** [comment]
Hi @hiirrxnn  i would like to confirm again that whether you are still working on the `UpdateColumnPosition` and `UpdateColumnType` mentioned in [#1565](https://github.com/datastrato/gravitino/issues/1565) or not, and looking forward to your reply. Thank you

**Comment by caican00** [comment]
i will work on this.

**Comment by hiirrxnn** [comment]
We can split the 2 subtasks and do one each!

**Comment by caican00** [comment]
> I was working on it ! Have you completed both the subtasks?

Hi @hiirrxnn i have created pull request for them. If you would like to keep doing it, I will close my pr.

**Comment by hiirrxnn** [comment]
No problem , you can continue on your PR , I will find other issues to work on .

**Comment by caican00** [comment]
> No problem , you can continue on your PR , I will find other issues to work on .

Maybe you could follow up on this issue if you're interested in it.
https://github.com/datastrato/gravitino/issues/2498

**Comment by caican00** [comment]
> > No problem , you can continue on your PR , I will find other issues to work on .
> 
> Maybe you could follow up on this issue if you're interested in it. cc @hiirrxnn #2498

Hi @hiirrxnn this feature is expected in 0.5 which is released in 4.16, is the time suitable for you?
